dspace-server-webapp/src/test/java/org/dspace/app/rest/ItemUploadControllerIT.java

```diff
@@ -0,0 +1,538 @@
+        Community child1 = CommunityBuilder.createSubCommunity(context, parentCommunity)
+                                           .withName("Sub Community")
+                                           .build();
+        Collection col1 = CollectionBuilder.createCollection(context, child1).withName("Collection 1").build();
+        Collection col2 = CollectionBuilder.createCollection(context, child1).withName("Collection 2").build();
+        Collection col3 = CollectionBuilder.createCollection(context, child1).withName("OrgUnits").build();
+
+        Item item = ItemBuilder.createItem(context, col1)
+                               .withTitle("Author1")
+                               .withIssueDate("2017-10-17")
+                               .withAuthor("Smith, Donald")
+                               .build();
+        context.setCurrentUser(eperson);
+        authorizeService.addPolicy(context, item, Constants.ADD, eperson);
+        authorizeService.addPolicy(context, item, Constants.WRITE, eperson);
+        String token = getAuthToken(eperson.getEmail(), password);
+
+        String input = "Hello, World!";
+
+        MockMultipartFile file = new MockMultipartFile("file", "hello.txt", MediaType.TEXT_PLAIN_VALUE,
+                                                       input.getBytes());
+
+        BitstreamPropertiesRest bitstreamPropertiesRest = new BitstreamPropertiesRest();
+        String testbundle = "TESTBUNDLE";
+        bitstreamPropertiesRest.setBundleName(testbundle);
+
+        ObjectMapper mapper = new ObjectMapper();
+
+
+        context.restoreAuthSystemState();
+        MvcResult mvcResult = getClient(token)
+            .perform(MockMvcRequestBuilders.fileUpload("/api/core/items/" + item.getID() + "/bitstreams")
+                                           .file(file)
+                                           .param("properties", mapper
+                                               .writeValueAsString(bitstreamPropertiesRest)))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.bundleName", is(testbundle)))
+            .andExpect(jsonPath("$.uuid", notNullValue())).andReturn();
+
+
+        String content = mvcResult.getResponse().getContentAsString();
+        Map<String, Object> map = mapper.readValue(content, Map.class);
+        String bitstreamId = String.valueOf(map.get("id"));
+
+        getClient(token).perform(get("/api/core/items/" + item.getID() + "/bitstreams"))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("_embedded.bitstreams", Matchers.hasItem(
+                            BitstreamMatcher.matchBitstreamEntry(UUID.fromString(bitstreamId), file.getSize()))));
+    }
+
+    // TODO This test doesn't work either as it seems that only the first file is ever transfered into the request
+    // Thus we cannot check for this and we have no way of knowing how many files we gave to the request
```

Comment:
This TODO seems to imply that we need to update our REST Contract?  Should we note in the Contract that this endpoint only accepts one file at a time...and if multiple files are sent, any extra files will be ignored?

Reply:
I guess that the issue is in the Spring binding that automatically strip out the extra files... probably if you access the original request and look for the file attribute manually you should be able to check how many file are here... or another option is to access an array of multipartfile in the method and just return error if the array has a size > 1.
Anyway, I'm also fine with the suggestion of @tdonohue to just note in the REST contract (and in the javadoc of the controller) that any extra files is silently ignored

Reply:
I've adjusted it in https://github.com/DSpace/Rest7Contract/pull/71 (can you please approve it, it's a minor change).
We'll also update the JavaDoc

Reply:
The JavaDoc has been updated